web/frontend/src/components/slam-2d-render.vue

```diff
@@ -42,27 +52,99 @@ scene.add(camera);
 const controls = new MapControls(camera, canvas);
 controls.enableRotate = false;
 
-const raycaster = new MouseRaycaster({ camera, renderer, recursive: false });
+const raycaster = new MouseRaycaster({ camera, canvas, recursive: false });
 
-raycaster.on('click', (event) => {
+raycaster.addEventListener('click', async (event) => {
   const [intersection] = event.intersections as THREE.Intersection[];
   if (intersection && intersection.point) {
-    console.log(intersection.point);
+    emit('click', intersection.point)
   }
 });
 
-const markerSize = 0.5;
-const marker = new THREE.Mesh(
-  new THREE.PlaneGeometry(markerSize, markerSize).rotateX(-Math.PI / 2),
-  new THREE.MeshBasicMaterial({ color: 'red' })
-);
-marker.name = 'Marker';
-// This ensures the robot marker renders on top of the pointcloud data
-marker.renderOrder = 999;
+
+const makeMarker = async (url : string, name: string, scalar: number) => {
+  const guiData = {
+    currentURL: url,
+    drawFillShapes: true,
+    drawStrokes: true,
+    fillShapesWireframe: false,
+    strokesWireframe: false
+  };
+
+  const svgLoader = new SVGLoader();
+  const data = await svgLoader.loadAsync(guiData.currentURL)
+
+  const { paths } = data!;
+
+  const group = new THREE.Group();
+  group.scale.multiplyScalar(scalar);
+  group.position.x = - 70;
+  group.position.y = 70;
+  group.scale.y *= - 1;
+
+  for ( let i = 0; i < paths.length; i ++ ) {
+
+    const path = paths[ i ];
+
+    const fillColor = path.userData!.style.fill;
+    
+    if ( guiData.drawFillShapes && fillColor !== undefined && fillColor !== 'none' ) {
+      const material = new THREE.MeshBasicMaterial( {
+        color: new THREE.Color().setStyle( fillColor ).convertSRGBToLinear(),
+        opacity: path.userData!.style.fillOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        depthWrite: false,
+        wireframe: guiData.fillShapesWireframe
+      } );
+
+      const shapes = SVGLoader.createShapes( path );
+
+      for ( let j = 0; j < shapes.length; j ++ ) {
+
+        const shape = shapes[ j ];
+
+        const geometry = new THREE.ShapeGeometry( shape );
+        const mesh = new THREE.Mesh( geometry.rotateX(-Math.PI / 2).rotateY(Math.PI), material );
+        group.add( mesh );
+
+      }
+
+    }
+
+    const strokeColor = path.userData!.style.stroke;
+
+    if ( guiData.drawStrokes && strokeColor !== undefined && strokeColor !== 'none' ) {
+
+      const material = new THREE.MeshBasicMaterial( {
+        color: new THREE.Color().setStyle( strokeColor ).convertSRGBToLinear(),
+        opacity: path.userData!.style.strokeOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        depthWrite: false,
+        wireframe: guiData.strokesWireframe
+      } );
+
+      for ( let j = 0, jl = path.subPaths.length; j < jl; j ++ ) {
+        const subPath = path.subPaths[ j ];
+        const geometry = SVGLoader.pointsToStroke( subPath.getPoints(), path.userData!.style );
```

Comment:
src/components/slam-2d-render.vue(130,73): error TS18048: 'path' is possibly 'undefined'

despite doing await at the top